dftimewolf/lib/state.py

```diff
@@ -813,3 +827,26 @@ def PublishMessage(self,
       message: The message content.
       is_error: True if the message is an error message, False otherwise."""
     self.cursesdm.EnqueueMessage(source, message, is_error)
+
+
+def FindCallerModule() -> str:
+  """Finds the caller module runtime name.
+
+  This method is used in Get/StoreContainer to find the module name that called
+  that method. Used for debug logging to track container lifecycles.
+
+  This method looks two frames up the stack for the caller information:
+  * 0: state.py:FindCallerModule (this frame when the method is called)
+  * 1: state.py:DFTimewolfState.Get/StoreContainer()
+  * 2: <module>.<method> <- The module name we want.
+
+  In unit tests, #2 is the unit test class, which has no 'name' attribute. In
+  such a scenario, fall back to the class name.
+
+  Returns:
+    A string of the calling module runtime name if present, else the class name.
+  """
+  desired_stack_frame = 2
+  caller_locals = inspect.stack()[desired_stack_frame][0].f_locals["self"]
+  return str((caller_locals.name if hasattr(caller_locals, 'name') else
```

Comment:
At this point wouldn't it be better to either make `name` a property in module.py, or write a short GetName() function? I remember getting bitten by similar things using runtime_name before.

Reply:
I can add that in, but it wouldn't really solve the problem with breaking the unit tests (which is why the `if hasattr(caller_locals, 'name')` is here) because those tests, which are two frames up the stack, don't have `name`, whether as an attribute or a property.

Reply:
Ah, you mean because we manually add modules in the unit tests. Agreed, let's not go with the property way then. Also I don't think we should find workarounds in our codebase just so that unit tests are happy; that's an early sign of code smell. Going with my comment in https://github.com/log2timeline/dftimewolf/pull/689#discussion_r1054072994 would make this problem go away, no? Since we could pass in the name in our call directly.

Reply:
Updated to have new methods in module Get/StoreContainer, and updated all modules and tests to use those methods instead.